server/utils/azureAppInsights.test.ts

```diff
@@ -0,0 +1,175 @@
+import { DataTelemetry, EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
+import { Contracts } from 'applicationinsights'
+import {
+  addUserDataToRequests,
+  ContextObject,
+  ignoredDependenciesProcessor,
+  ignoredRequestsProcessor,
+} from './azureAppInsights'
+import { HmppsUser, PrisonUser } from '../interfaces/hmppsUser'
+
+const exampleUser = {
+  username: 'test-user',
+  authSource: 'nomis',
+} as HmppsUser
+
+const createEnvelope = (properties: Record<string, string | boolean>, baseType = 'RequestData') =>
+  ({
+    data: {
+      baseType,
+      baseData: { properties },
+    } as DataTelemetry,
+  }) as EnvelopeTelemetry
+
+const createContext = (user: HmppsUser) =>
+  ({
+    'http.ServerRequest': {
+      res: {
+        locals: {
+          user,
+        },
+      },
+    },
+  }) as ContextObject
+
+const context = createContext(exampleUser)
+
+describe('azureAppInsights', () => {
+  describe('addUserDataToRequests', () => {
+    it('adds user data to properties when present', () => {
+      const envelope = createEnvelope({ other: 'things' })
+
+      addUserDataToRequests(envelope, context)
+
+      expect(envelope.data.baseData.properties).toStrictEqual({
+        username: exampleUser.username,
+        authSource: exampleUser.authSource,
+        other: 'things',
+      })
+    })
+
+    it('adds activeCaseLoadId to properties for a prison user with this set', () => {
+      const envelope = createEnvelope({ other: 'things' })
+
+      addUserDataToRequests(envelope, createContext({ ...exampleUser, activeCaseLoadId: 'MDI' } as PrisonUser))
+
+      expect(envelope.data.baseData.properties).toStrictEqual({
+        username: exampleUser.username,
+        authSource: exampleUser.authSource,
+        activeCaseLoadId: 'MDI',
+        other: 'things',
+      })
+    })
+
+    it.each(['delius', 'external', 'azuread'])('handles %s users', (authSource: 'delius' | 'external' | 'azuread') => {
```

Comment:
Technically we won't have these type of users.  However it does no harm keeping this here.